service/src/test/scala/com/yahoo/maha/service/example/MahaServiceExampleTest.scala

```diff
@@ -321,3 +327,1196 @@ class MahaServiceExampleTest extends BaseMahaServiceTest with Logging with Befor
+  }
+
+  // Need fix: dim table where condition always has an empty filter as the first filter; random join order
+  ignore("Test: fact table join with 2 same dim level tables in Hive should succeed") {
+    val jsonString =
+      s"""
+         |{
+         |    "cube": "student_performance",
+         |    "selectFields": [
+         |        {
+         |            "field": "Student Name"
+         |        },
+         |        {
+         |            "field": "Researcher Name"
+         |        },
+         |        {
+         |            "field": "Researcher Status"
+         |        },
+         |        {
+         |            "field": "Marks Obtained"
+         |        }
+         |    ],
+         |    "filterExpressions": [
+         |        {
+         |            "field": "Day",
+         |            "operator": "between",
+         |            "from": "$fromDate",
+         |            "to": "$toDate"
+         |        },
+         |        {
+         |            "field": "Student ID",
+         |            "operator": "=",
+         |            "value": "213"
+         |        },
+         |        {
+         |            "field": "Tutor Status",
+         |            "operator": "=",
+         |            "value": "admitted"
+         |        }
+         |    ]
+         |}
+         |""".stripMargin
+
+    val request: ReportingRequest = ReportingRequest.forceHive(getReportingRequestAsync(jsonString, StudentSchema))
+    val registry = exampleRegistry
+    val res = getRequestModel(request, registry, revision = Some(3))
+    assert(res.isSuccess, s"Building request model failed.")
+
+    val queryPipelineTry = generatePipeline(res.toOption.get)
+    assert(queryPipelineTry.isSuccess, queryPipelineTry.errorMessage("Fail to get the query pipeline"))
+
+    val queryPipeline = queryPipelineTry.toOption.get
+    val result = queryPipeline.queryChain.drivingQuery.asString
+    println(result)
+
+    val expected =
+      s"""
+         |SELECT CONCAT_WS(",",NVL(CAST(mang_student_name AS STRING), ''), NVL(CAST(mang_researcher_name AS STRING), ''), NVL(CAST(mang_researcher_status AS STRING), ''), NVL(CAST(mang_marks_obtained AS STRING), ''))
+         |FROM(
+         |SELECT COALESCE(s3.mang_student_name, "NA") mang_student_name, COALESCE(r2.mang_researcher_name, "NA") mang_researcher_name, COALESCE(r2.mang_researcher_status, "NA") mang_researcher_status, COALESCE(obtained_marks, 0L) mang_marks_obtained
+         |FROM(SELECT tutor_id, student_id, researcher_id, SUM(obtained_marks) obtained_marks
+         |FROM hive_student_performance
+         |WHERE (student_id = 213) AND (date >= '$fromDateHive' AND date <= '$toDateHive')
+         |GROUP BY tutor_id, student_id, researcher_id
+         |
+         |       )
+         |hsp0
+         |JOIN (
+         |SELECT id t1_id
+         |FROM hive_tutor
+         |WHERE (()) AND (status = 'admitted')
```

Comment:
use test partitionDimensionColumnRenderer:PartitionColumnRenderer, like the one we use in HiveQueryGeneratorV2Test